src/app.js

```diff
@@ -51,13 +56,18 @@ app.post('/messages/:channel', (req, res, next) => {
    *
    * @return {Function}
    */
-function publishJob(channel, messages, end) {
+function publishJob(channel, messages, end, retries=0) {
   return function(cb) {
     pubsub.publish(channel, messages).then((result)=>{
       let errors = result.errors;
       if(errors.length > 0) {
         prom.publishCount.inc({state: 'failed', channel});
-        queue.push(publishJob(channel, errors, end));
+        retries++;
+        if(config.FALLBACK && retries >= 2) {
+          end();
+          return rabbit.publish(channel, errors);
+        }
+        queue.push(publishJob(channel, errors.map((error) => error.message), end, retries));
```

Comment:
Not sure how this stuff works (particularly `end();`) but it looks as if this will continue to requeue the event, increasing the retries count ad infinitum, and publish a new rabbit job each time where retries > 2?

Reply:
end is a function from prometheus (it ends the timer on the response time percentiles)
The queue.push won't be hit when retries are >= 2 because of the return statement. So it will publish the rabbit message and end the last job.

Reply:
:+1:  cool somehow wasn't reading the `return`